clang/test/SemaSYCL/sycl-cuda-compilation.cu

```diff
@@ -0,0 +1,67 @@
+// RUN: %clang_cc1 %s -fsycl-is-host -D__sycl_cuda_host \
+// RUN:   -internal-isystem %S/../SemaCUDA/Inputs \
+// RUN:   -fsycl-targets=nvptx64-nvidia-cuda -triple x86_64-unknown-linux \
+// RUN:   -emit-llvm -o - -verify -verify-ignore-unexpected=note
+// RUN: %clang_cc1 %s -fsycl-is-host -fcuda-is-device -D__cuda_device \
+// RUN:   -internal-isystem %S/../SemaCUDA/Inputs \
+// RUN:   -fsycl-targets=nvptx64-nvidia-cuda -triple x86_64-unknown-linux \
+// RUN:   -emit-llvm -o - -verify -verify-ignore-unexpected=note
+// RUN: %clang_cc1 %s -fsycl-is-device -D__sycl_device \
+// RUN:   -internal-isystem %S/../SemaCUDA/Inputs \
+// RUN:   -fsycl-targets=nvptx64-nvidia-cuda -triple x86_64-unknown-linux\
+// RUN:   -emit-llvm -o - -verify -verify-ignore-unexpected=note
+
+// This tests the errors emitted by SEMA in case of SYCL-CUDA compilation.
+
+// | F  | T  |     |
+// |----+----+-----+
+// | d  | h  | (A) |
+// | g  | h  | (B) |
+// | g  | g  | (C) |
+// | d  | g  | (D) |
+// | hd | g  | (E) |
+
+#include "cuda.h"
+
```

Comment:
Small refactoring suggestion.
Tests cases duplicate some declarations, so we can move them to the common section.
```c++
__host__ void host_fn() {}
__device__ void dev_fn {}
__global__ void global_fn() {}
```
We can use empty functions as building blocks for test cases below.